doc: Add migration guide for LwM2M related changes (#12671)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.5.rst

The following changes are recommended for your application to work optimally after the migration.

> * Latest changes from Zephyr and nRF Connect SDK allows power optimization for LwM2M client.
>   Use of DTLS Connection Identifier reduces the DTLS handshake overhead when doing the LwM2M Update operation.
>   This is enabled using :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` and requires modem firmware 1.3.5 or newer.
>   Zephyr's LwM2M engine now support tickless operation mode when :kconfig:option:`CONFIG_LWM2M_TICKLESS` is enabled.
>   This prevents device from waking up on every 500 ms and achieves longer sleep periods.
>   These power optimizations are enabled on our LwM2M sample when :file:`overlay-dtls-cid.conf` is used.

Reviewer: ```suggestion
* Latest changes in Zephyr and nRF Connect SDK allow power optimization for the LwM2M client.
  Using DTLS Connection Identifier reduces the DTLS handshake overhead when performing the LwM2M Update operation.
  This is enabled using the :kconfig:option:`CONFIG_LWM2M_CLIENT_UTILS_DTLS_CID` Kconfig option and requires modem firmware v1.3.5 or newer.
  Zephyr's LwM2M engine now support tickless operation mode when the Kconfig option :kconfig:option:`CONFIG_LWM2M_TICKLESS` is enabled.
  This prevents the device from waking up on every 500 ms and achieves longer sleep periods.
  These power optimizations are enabled on the :ref:`lwm2m_client` sample when using the :file:`overlay-dtls-cid.conf` overlay file.
```
Which sample, client or

> * Applications that use Zephyr's LwM2M stack and :ref:`lib_lwm2m_client_utils` library should refactor to use new event :c:member:`LWM2M_FOTA_UPDATE_MODEM_RECONNECT_REQ` when doing modem firmware update. See :ref:`lwm2m_client` sample for example.

Reviewer 2: when modem is updated and initilized and ready for re-connect

Author: That is too much details.
This event belongs to a FOTA process. Rest of the details are in example application and documentation.

> * Applications that use Zephyr's LwM2M stack and :ref:`lib_lwm2m_client_utils` library should refactor to use new event :c:member:`LWM2M_FOTA_UPDATE_MODEM_RECONNECT_REQ` when doing modem firmware update to avoid rebooting the device. See :ref:`lwm2m_client` sample for example.

Reviewer 2: No it is good

> * Applications that use Zephyr's LwM2M stack and :ref:`lib_lwm2m_client_utils` library should refactor to use new event :c:member:`LWM2M_FOTA_UPDATE_MODEM_RECONNECT_REQ` when doing modem firmware update to avoid rebooting the device. See :ref:`lwm2m_client` sample for example.

Reviewer: ```suggestion
* Applications that use Zephyr's LwM2M stack and the :ref:`lib_lwm2m_client_utils` library must refactor to use the new event :c:member:`LWM2M_FOTA_UPDATE_MODEM_RECONNECT_REQ` when updating the modem firmware to avoid rebooting the device.
  For an example, see the :ref:`lwm2m_client` sample.
```
